server/server_test.go

```diff
@@ -560,6 +562,47 @@ var _ = Describe("Server package test", func() {
 				})
 			})
 
+			Context("Same tenancy condition", func() {
+				const (
+					relatedResourceID = "some_id1"
+					mainResourceID    = "some_id2"
+				)
+
+				BeforeEach(func() {
+					relatedResource := map[string]interface{}{
+						"id":        relatedResourceID,
+						"tenant_id": powerUserTenantID,
+					}
+					testURL(http.MethodPost, sameTenancyRelatedPluralURL, adminTokenID, relatedResource, http.StatusCreated)
+				})
+
+				It("Should not create resource with different tenancy", func() {
+					mainResource := map[string]interface{}{
+						"id":                               mainResourceID,
+						"same_tenancy_related_resource_id": relatedResourceID,
+						"tenant_id":                        memberTenantID,
+					}
+					testURL(http.MethodPost, sameTenancyMainPluralURL, adminTokenID, mainResource, http.StatusBadRequest)
```

Comment:
For "update" we test both same and different tenancy scenarios. Why not also for "create"?

Reply:
The test was added.